### PR TITLE
CAMS-442 update mongo keys and refactor case assigment

### PR DESCRIPTION
**Pull request by amorrow-flexion**
Jira ticket: CAMS-442

# Problem

Offices collection in the cosmos repository has a unique key policy on office_code which is causing errors in the office-staff-sync

# Solution

removed unique key policy from offices collection in bicep

# Testing/Validation

- *deployed to environment

# Other Changes

N/A

### CODE REVIEW

File: backend/functions/lib/adapters/gateways/mongo/utils/mongo-adapter.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export class MongoCollectionAdapter<T> implements DocumentCollectionAdapter<T> {
 
   public async find(query: ConditionOrConjunction, sort?: Sort): Promise<T[]> {
     const mongoQuery = toMongoQuery(query);
+    console.log(mongoQuery);
```
Comment (btposey) [inline backend/functions/lib/adapters/gateways/mongo/utils/mongo-adapter.ts:32]:
```suggestion
```

File: backend/functions/lib/use-cases/case-management.test.ts [hunk 2]
```diff
@@ -214,6 +216,10 @@ describe('Case management tests', () => {
         return Promise.resolve(caseDetail);
       });
 
+      jest.spyOn(chapterCaseList.casesGateway, 'getCaseDetail').mockImplementation(async () => {
+        return Promise.resolve(caseDetail);
+      });
```
Comment (btposey) [inline backend/functions/lib/use-cases/case-management.test.ts:221]:
```suggestion
      jest.spyOn(chapterCaseList.casesGateway, 'getCaseDetail').mockResolvedValue(caseDetail);
```

File: backend/functions/lib/use-cases/case-management.ts [hunk 3]
```diff
@@ -69,19 +70,37 @@ export default class CaseManagement {
...
       for (const casesKey in cases) {
+        caseIds.push(cases[casesKey].caseId);
         const bCase = cases[casesKey];
         bCase.officeCode = buildOfficeCode(bCase.regionId, bCase.courtDivisionCode);
         bCase._actions = getAction<CaseBasics>(context, bCase);
       }
+
+      if (includeAssignments) {
+        const assignmentsMap = await this.assignmentGateway.findAssignmentsByCaseId(caseIds);
+        console.log('Assignments', JSON.stringify(assignmentsMap));
```
Comment (btposey) [inline backend/functions/lib/use-cases/case-management.ts:94]:
```suggestion
```

File: backend/functions/lib/use-cases/orders/orders-consolidation-approval.test.ts [hunk 4]
```diff
@@ -150,7 +150,23 @@ describe('Orders use case', () => {
...
       });
 
-    jest.spyOn(MockMongoRepository.prototype, 'findAssignmentsByCaseId').mockResolvedValue([]);
+    jest.spyOn(MockMongoRepository.prototype, 'update').mockImplementation(() => {
+      return Promise.resolve();
+    });
+
+    jest.spyOn(MockMongoRepository.prototype, 'create').mockImplementation(() => {
+      return Promise.resolve();
+    });
```
Comment (btposey) [inline backend/functions/lib/use-cases/orders/orders-consolidation-approval.test.ts:159]:
```suggestion
    jest.spyOn(MockMongoRepository.prototype, 'update').mockResolvedValue();

    jest.spyOn(MockMongoRepository.prototype, 'create')..mockResolvedValue();
```

File: user-interface/src/case-detail/CaseDetailScreen.tsx [hunk 5]
```diff
@@ -317,9 +319,18 @@ export default function CaseDetailScreen(props: CaseDetailProps) {
...
 
   function handleCaseAssignment(assignment: CallbackProps) {
+    const assignments: CaseAssignment[] = [];
+    assignment.selectedAttorneyList.forEach((attorney) => {
+      assignments.push({
+        userId: attorney.id,
+        name: attorney.name,
+        role: CamsRole.TrialAttorney,
+        assignedOn: new Date().toString(),
+      } as CaseAssignment);
```
Comment (btposey) [inline user-interface/src/case-detail/CaseDetailScreen.tsx:329]: I don't like this but I don't have a better suggestion at the moment.

**Review by btposey — CHANGES_REQUESTED** [review]

File: backend/functions/lib/controllers/cases/cases.controller.ts [hunk 6]
```diff
@@ -32,7 +32,7 @@ export class CasesController implements CamsController {
       if (context.request.method === 'GET' && context.request.params.caseId) {
         data = await this.getCaseDetails({ caseId: context.request.params.caseId });
       } else {
-        data = await this.searchCases(context.request);
+        data = await this.searchCases(context.request, context.request.query ?? {});
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/cases/cases.controller.ts:35]: This seems kind of odd if we are already sending context.request, then why also send context.request.query?

File: user-interface/src/case-detail/CaseDetailScreen.tsx [hunk 7]
```diff
@@ -317,9 +319,18 @@ export default function CaseDetailScreen(props: CaseDetailProps) {
   }
 
   function handleCaseAssignment(assignment: CallbackProps) {
+    const assignments: CaseAssignment[] = [];
+    assignment.selectedAttorneyList.forEach((attorney) => {
+      assignments.push({
+        userId: attorney.id,
+        name: attorney.name,
+        role: CamsRole.TrialAttorney,
+        assignedOn: new Date().toString(),
```
Comment (btposey) [inline user-interface/src/case-detail/CaseDetailScreen.tsx:328]:
```suggestion
```

**Review by btposey — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous-deployment.yml [hunk 8]
```diff
@@ -125,11 +125,6 @@ jobs:
       [
         setup,
         build,
-        accessibility-test,
-        security-scan,
-        unit-test-frontend,
-        unit-test-backend,
-        unit-test-common,
```
Comment (jamesobrooks) [inline .github/workflows/continuous-deployment.yml:132]: We need to put these back.

**Review by btposey — APPROVED** [review]

File: backend/functions/lib/controllers/cases/cases.controller.ts (hunk 6 above)
Comment (jamesobrooks) [reply backend/functions/lib/controllers/cases/cases.controller.ts:35]: We can certainly change this.

**Review by fmaddenflx — COMMENTED** [review]
Looks good overall

**Pull request merged**